docsite/rst/developing_program_flow_modules.rst

```diff
@@ -0,0 +1,392 @@
+.. _flow_modules:
+
+=======
+Modules
+=======
+
+This is an in-depth dive into understanding Ansible's program flow to execute
+modules.  It will be of use to people working on the portions of the Core
+Ansible Engine that execute a module, it may be of interest to people writing
+Ansible Modules, and people simply wanting to use Ansible Modules will likely
+not find anything helpful here.
+
```

Comment:
Just wordsmithing for most of these editing suggestions....
This in-depth dive helps you understand Ansible's program flow to execute modules. It is written for people working on the portions of the Core Ansible Engine that execute a module.  Those writing Ansible Modules may also find this in-depth dive to be of interest, but individuals simply using Ansible Modules will not likely find this to be helpful.